modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/JobRepository.java

```diff
@@ -0,0 +1,181 @@
+        JobEntity jobEntity = mapper.map(jobModel, JobEntity.class);
+
+        if (jobEntity.getJobStatuses() != null) {
+            logger.debug("Populating the Primary Key of JobStatus objects for the Job");
+            jobEntity.getJobStatuses().forEach(jobStatusEntity -> {
+                jobStatusEntity.setJobId(jobId);
+                jobStatusEntity.setTaskId(taskId);
+            });
+        }
+
+        if (!isJobExist(cis)) {
+            logger.debug("Checking if the Job already exists");
+            jobEntity.setCreationTime(new Timestamp((System.currentTimeMillis())));
+        }
+
+        return execute(entityManager -> entityManager.merge(jobEntity));
+    }
+
+    public String addJob(JobModel job, String processId) throws RegistryException {
+        CompositeIdentifier cis = new CompositeIdentifier(job.getTaskId(), job.getJobId());
+        String jobId = saveJobModelData(job, cis);
+        String taskId = (String) cis.getTopLevelIdentifier();
+        job.setProcessId(processId);
+        job.setTaskId(taskId);
+        TaskRepository taskRepository = new TaskRepository();
+        TaskModel taskModel = taskRepository.getTask(taskId);
+        List<JobModel> jobModelList = taskModel.getJobs();
+
+        if (jobModelList != null && !jobModelList.contains(job)) {
+            logger.debug("Adding the Job to the list");
+            jobModelList.add(job);
+            taskModel.setJobs(jobModelList);
+            taskRepository.updateTask(taskModel, taskId);
+        }
+
+        return jobId;
+    }
+
+    public String updateJob(JobModel job, CompositeIdentifier cis) throws RegistryException {
+        return saveJobModelData(job, cis);
+    }
+
+    public JobModel getJob(CompositeIdentifier cis) throws RegistryException {
+        JobRepository jobRepository = new JobRepository();
+        return jobRepository.get((String) cis.getSecondLevelIdentifier());
+    }
+
+    public String addJobStatus(JobStatus jobStatus, CompositeIdentifier cis) throws RegistryException {
+        JobModel jobModel = getJob(cis);
+        List<JobStatus> jobStatusList = jobModel.getJobStatuses();
+
+        if (jobStatusList.size() == 0 || !jobStatusList.contains(jobStatus)) {
+
+            if (jobStatus.getStatusId() == null) {
+                logger.debug("Set JobStatus's StatusId");
+                jobStatus.setStatusId(ExpCatalogUtils.getID("STATUS"));
+            }
+
+            logger.debug("Adding the JobStatus to the list");
+            jobStatusList.add(jobStatus);
+
+        }
+
+        jobModel.setJobStatuses(jobStatusList);
+        updateJob(jobModel, cis);
+        return jobStatus.getStatusId();
+    }
+
+    public String updateJobStatus(JobStatus updatedJobStatus, CompositeIdentifier cis) throws RegistryException {
+        JobModel jobModel = getJob(cis);
+        List<JobStatus> jobStatusList = jobModel.getJobStatuses();
+
+        for (JobStatus retrievedJobStatus : jobStatusList) {
```

Comment:
@tilaks26 just a thought, did you consider creating a JobStatusRepository and then updating them directly? This approach where you loop through a list to find and update a JobStatus and then you update the entire Job just seems inefficient and complicates the code.